src/modules/users/sagas/user.ts

```diff
@@ -209,10 +219,157 @@ function* userLogout() {
   return null;
 }
 
+function* userDepositToken({
+  meta,
+  payload: { tokenAddress, amount, colonyAddress },
+}: Action<ActionTypes.USER_DEPOSIT_TOKEN>) {
+  const txChannel = yield call(getTxChannel, meta.id);
+  try {
+    const apolloClient = TEMP_getContext(ContextModule.ApolloClient);
+    const colonyManager = TEMP_getContext(ContextModule.ColonyManager);
+    const { walletAddress } = yield getLoggedInUser();
+
+    // eslint-disable-next-line max-len
+    const tokenLockingClient: TokenLockingClient = yield colonyManager.getClient(
+      ClientType.TokenLockingClient,
+      colonyAddress,
+    );
+
+    const batchKey = 'deposit';
+
+    const { approve, deposit } = yield createTransactionChannels(meta.id, [
+      'approve',
+      'deposit',
+    ]);
+
+    const createGroupTransaction = ({ id, index }, config) =>
+      fork(createTransaction, id, {
+        ...config,
+        group: {
+          key: batchKey,
+          id: meta.id,
+          index,
+        },
+      });
+
+    yield createGroupTransaction(approve, {
+      context: ClientType.TokenClient,
+      methodName: 'approve',
+      identifier: colonyAddress,
+      params: [tokenLockingClient.address, amount],
+      ready: false,
+    });
+
+    yield createGroupTransaction(deposit, {
+      context: ClientType.TokenLockingClient,
+      methodName: 'deposit',
+      identifier: colonyAddress,
+      params: [tokenAddress, amount, false],
+      ready: false,
+    });
```

Comment:
The amount value needs to be converted to a big number.

Otherwise, for values with more than 15 zeros. this will give an error.

So something like:
```
params: [tokenAddress, new BigNumber(amount), false],
```

This applies to both approve and deposit